sklearn/decomposition/tests/test_sparse_pca.py

```diff
@@ -79,6 +79,26 @@ def test_fit_transform():
                          Y, ridge_alpha=None)
 
 
+def test_fit_transform_variance():
+    alpha = 1
+    rng = np.random.RandomState(0)
+    Y, _, _ = generate_toy_data(3, 10, (8, 8), random_state=rng)  # wide array
+    spca_lars = SparsePCA(n_components=3, method='lars', alpha=alpha,
+                          random_state=0, variance=True)
+    pca = PCA(n_components=3, random_state=0)
+
+    pca.fit(Y)
+    # no need to fit spca for this
+    spca_lars.fit(Y)
+
+    components = pca.components_
+    explained_variance = pca.explained_variance_
+    spca_lars.components_ = components
+    explained_variance_sparse = spca_lars.explained_variance_
+
```

Comment:
hum, I misunderstood what you're doing. 
You just want to assert that your function `̀_get_explained_variance` returns the same result as in PCA if given same X and same components, right ?
Maybe you can add a comment before line 96 explaining that.

And I guess you have to do `explained_variance_sparse = spca_lars._get_explained_variance_(Y)`

Is it possible to make a test checking that the explained_variance_ computed with the components fitted by SparsePCA are correct ? maybe something like comparing explained_variance_ between PCA and SparsePCA with sparsity=0 ?